undertow/src/main/java/org/wildfly/extension/undertow/filters/ModClusterDefinition.java

```diff
@@ -322,29 +333,81 @@ private ModClusterDefinition() {
     @Override
     public HttpHandler createHttpHandler(Predicate predicate, ModelNode model, HttpHandler next) {
         throw new IllegalStateException(); //this is not used for mod_cluster, as it required injection and socket binding
+    }
 
+    @Override
+    public ManagementResourceRegistration register(ManagementResourceRegistration parent) {
+        ResourceDescriptor descriptor = new ResourceDescriptor(this.getResourceDescriptionResolver())
+                .addCapabilities(Capability.class)
+                .addRequiredSingletonChildren(SingleAffinityResourceDefinition.PATH)
+                ;
+
+        for (AttributeDefinition attribute : ATTRIBUTES) {
+            descriptor.addAttributes(new Attribute() {
+                @Override
+                public AttributeDefinition getDefinition() {
+                    return attribute;
+                }
+            });
+        }
+
+        ModClusterResourceServiceHandler handler = new ModClusterResourceServiceHandler();
+        new ModClusterResourceRegistration(descriptor, handler).register(parent);
+
+        new NoAffinityResourceDefinition().register(parent);
+        new SingleAffinityResourceDefinition().register(parent);
+        new RankedAffinityResourceDefinition().register(parent);
+
+        parent.registerSubModel(ModClusterBalancerDefinition.INSTANCE);
```

Comment:
These are child resources, no?  Should you register them using the MRR of the ModClusterResourceDefinition?

Reply:
No.

But as discussed on the call I have reworked this.